### PR TITLE
[LOCATION] [9.12.r1] Forwardport commits from 8.1.r1 and fix build

### PR DESCRIPTION
**Pull request by kholk**
This is a forward port of relevant commits from the old 8.1.r1 branch.
It also contains a fix for one of the reversed structs in the qmi headers.

Build Test on Android11: OK

### CODE REVIEW

**Comment by kholk** [comment]
This PR now builds fine and can be merged.

**Comment by MarijnS95** [comment]
> I know that somebody surely has it fixed locally (@MarijnS95 hello!), so I'm not wasting time on this.

Sorry to burst your bubble, but noupe, `-ENOTIME`!

Anyway this builds fine and instantly gives a bunch of satellites on PDX206 (that have previously been successfully acquired on 8.1 branches). Waiting for a fix now, but so far so good, thanks for sorting through that massive list of patches!

**Comment by MarijnS95** [comment]
And I just did something completely unholy, because my tree is currently set up for Edo/9.12/9.15 branches so I cannot build a clean vendor image for older devices. However, I can do this on Seine/PDX201:

```console
$ adb shell rm /vendor/lib*/{libloc,libgps}*
$ m libloc_core libgps.utils liblocation_api libbatching libgeofencing libgnss libloc_api_v02 libgnsspps libsynergy_loc_api android.hardware.gnss@2.0-impl-qti
#### build completed successfully (50 seconds) ####
$ adb sync
/vendor/: 12 files pushed, 10 skipped. 9.9 MB/s (1715956 bytes in 0.166s) # NOTE! Skipped because I already synced before, and added `gnss@2.0` in a second run which was initially forgotten about. 2 pushed files for 32- and 64-bit.
$ adb reboot
```

The device just found one satellite, and `dumpsys location` shows them all. It usually takes a bit of time, but this indicates we're golden on older platforms, as expected because nothing changed API (nor blob) wise.

**Comment by MarijnS95** [comment]
Good news: Seine/PDX201 saw about ~10 sats and already has a fix, Edo/PDX206 is still trying to figure out whether to connect, with 43 known satellites :grimacing:

**Comment by MarijnS95** [comment]
And finally, Seine is now connected to 11 sats while Edo is connected to 8. All is working fine.

File: utils/loc_socket/Android.mk [hunk 1]
```diff
@@ -36,8 +33,13 @@ LOCAL_CFLAGS += $(GNSS_CFLAGS)
 
 ifeq ($(TARGET_KERNEL_VERSION),$(filter $(TARGET_KERNEL_VERSION),3.18 4.4 4.9))
```
Comment (ix5) [inline utils/loc_socket/Android.mk:34]: Drop this whole commit, we're not using 4.9 kernels with this HAL

**Review by ix5 — CHANGES_REQUESTED** [review]

**Pull request merged**